### ISSUE TITLE
(aws_stepfunctions_tasks): (Incorrect service name in generated IAM role for step function excecution)

### ISSUE THREAD

**Issue by laura-russo-ocr** [aws/aws-cdk#23824]
### Describe the bug

When using the `CallAwsService` construct to call the startExecution action, the automatically generated IAM policy has `sfn` as the service name instead of the expected `states` service name

### Expected Behavior

Expected the IAM policy to have the correct service name, eg

`"Action": "states:startExecution"`

### Current Behavior

IAM policy instead has the incorrect service name e.g
`"Action": "sfn:startExecution"`

### Reproduction Steps

```
    const executeStateMachine = new sfntasks.CallAwsService(this, "executeStateMachine", {
      service: sfn,
      action: "startExecution",
      parameters: {
        StateMachineArn: stateMachine.stateMachineArn,
        Input: {
          "StatePayload": "Hello from Step Functions!",
          "AWS_STEP_FUNCTIONS_STARTED_BY_EXECUTION_ID.$": "$$.Execution.Id"
        }
      },
      iamResources: [*]
    });
```

### Possible Solution

CallAwsServiceProps state that `iamResources` will be set as `service:action` by default
Assuming the `service` referenced here is the `service` value in the props it will be incorrect in the case of the step function service

### CDK CLI Version

2.61.1 (build d319d9c)

### Node.js Version

v16.17.1

### OS

macOS Monterey

### Language

Typescript

### Other information

Thanks for your consideration, this is the first time I've logged a bug anywhere so hopefully I given the correct information.

**Comment by laura-russo-ocr** [comment]
Ok, I see it can also be worked around with `iamAction`
e.g 
```
        const getEmailIdentities = new sfntasks.CallAwsService(scope, 'getEmailIdentities', {
            service: 'sesv2',
            action: 'getEmailIdentity',
            parameters: {
                "EmailIdentity": 'test.com
            },
            iamResources: ['*'],
            iamAction: 'ses:GetEmailIdentity'
        });
```

Would suggest the doco be maybe updated to clarify that both the action and service can be defined here.

**Comment by pahud** [comment]
Yes this could be confusing but `iamAction` should be defined in this case. If you are interested to add some note in the doc, we'd be happy to review your PR submission. Thanks.

**Comment by vgkowski** [comment]
Same thing with EMR serverless: the service name needs to be `emrserverless` but the IAM policy needs `emr-serverless:*`

**Comment by AlJohri** [comment]
Running into this same issue as @vgkowski with EMR Serverless + SFN Tasks:

By default, CFN Template renders:

```
      {
       "Action": "emrserverless:startJobRun",
       "Effect": "Allow",
       "Resource": {
        "Fn::GetAtt": [
         "EmrServerless",
         "Arn"
        ]
       }
      },
```

When it should actually be `emr-serverless:startJobRun`

**Comment by gakinson** [comment]
+1 on the sesV2 vs ses. Using iamAction for now works for getting around it.
